Dockerfile

```diff
@@ -12,13 +12,13 @@ RUN apk -U upgrade \
     && wget -q https://storage.googleapis.com/kubernetes-release/release/v${KUBE_VERSION}/bin/${TARGETOS}/${TARGETARCH}/kubectl -O /usr/local/bin/kubectl \
     && wget -q https://get.helm.sh/helm-v${HELM_VERSION}-${TARGETOS}-${TARGETARCH}.tar.gz -O - | tar -xzO ${TARGETOS}-${TARGETARCH}/helm > /usr/local/bin/helm \
     && wget -q https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_${TARGETOS}_${TARGETARCH} -O /usr/local/bin/yq \
-    && chmod +x /usr/local/bin/helm /usr/local/bin/kubectl /usr/local/bin/yq \
-    && mkdir /config \
-    && chmod g+rwx /config /root \
-    && helm repo add "stable" "https://charts.helm.sh/stable" --force-update \
+    && chmod +x /usr/local/bin/helm /usr/local/bin/kubectl /usr/local/bin/yq
+
+RUN addgroup helm && adduser -h /config -D -G helm helm
```

Comment:
Having multiple RUN commands will make the image bigger

Reply:
It's impossible to execute all commands in one RUN here because "helm repo add" must run as user "helm". The other installations must run as user root, though.

Reply:
There *might* be people using `kubectl` with this image too - I'd make sure this and other tools install function.

Reply:
Thank you for your feedback!
Of course all tools, `kubectl`, `helm`, and `yq` still work. The username is arbitrary.
Because `helm repo add` writes to `~/.cache` and `~/.config`, it must be run with the target user instead of root.
`kubectl` and `yq` are not affected by this change at all because they are just binaries (owned by root.)

Are there still any objections? Is the additional layer a big problem? I think the security is worth the additional layers. Both images, before and after my change, are 152MB.

Reply:
This is a breaking change though, anyone using this image to generate charts/etc will now get files with a different UID/GID. The new image wont be able to read existing files either.

I rather see a separate Dockerfile that gets tagged as `-nonroot`

Reply:
You are right. My change will probably break things.
The tag `-nonroot` is a good idea. I will propose a solution for that, O.K.?

Reply:
@migu Yes

Reply:
@migu Any updates on this?